src/modules/InverseOrders.ts

```diff
@@ -0,0 +1,107 @@
+ * - The fill amount is denoted in quote currency (position).
+ * - Prices are denoted in quote currency per unit of base currency (position per margin).
+ * - The fee is paid (or received) in base currency (margin).
+ */
+export class InverseOrders extends Orders {
+  constructor(
+    contracts: Contracts,
+    web3: Web3,
+  ) {
+    super(contracts, web3, EIP712_DOMAIN_NAME, contracts.p1InverseOrders);
+  }
+
+  /**
+   * Estimate the maker's collateralization after executing a sequence of orders.
+   *
+   * The `maker` of every order must be the same. This function does not make any on-chain calls,
+   * so all information must be passed in, including the oracle price and remaining amounts
+   * on the orders. Orders are assumed to be filled at the limit price and limit fee.
+   *
+   * Returns the ending collateralization ratio for the account, or BigNumber(Infinity) if the
+   * account does not end with any negative balances.
+   *
+   * @param  initialBalance  The initial margin and position balances of the maker account.
+   * @param  oraclePrice     The price at which to calculate collateralization.
+   * @param  orders          A sequence of orders, with the same maker, to be hypothetically filled.
+   * @param  fillAmounts     The corresponding fill amount for each order, denominated in the token
+   *                         spent by the maker--quote currency when buying, and base when selling.
+   */
+  public getAccountCollateralizationAfterMakingOrders(
+    initialBalance: Balance,
+    oraclePrice: Price,
+    orders: Order[],
+    makerTokenFillAmounts: BigNumber[],
+  ): BigNumber {
+    const runningBalance: Balance = initialBalance.copy();
+
+    // For each order, determine the effect on the balance by following the smart contract math.
+    for (let i = 0; i < orders.length; i += 1) {
+      const order = orders[i];
+
+      const fillAmount = order.isBuy
+        ? makerTokenFillAmounts[i]
+        : makerTokenFillAmounts[i].times(order.limitPrice.value);
+
+      // Assume orders are filled at the limit price and limit fee.
+      const { marginDelta, positionDelta } = this.getBalanceUpdatesAfterFillingOrder(
+        fillAmount,
+        order.limitPrice,
+        order.limitFee,
+        order.isBuy,
+      );
+
+      runningBalance.margin = runningBalance.margin.plus(marginDelta);
+      runningBalance.position = runningBalance.position.plus(positionDelta);
+    }
+
+    return runningBalance.getCollateralization(oraclePrice);
+  }
+
+  /**
+   * Calculate the effect of filling an order on the maker's balances.
+   */
+  public getBalanceUpdatesAfterFillingOrder(
+    fillAmount: BigNumberable,
+    fillPrice: Price,
+    fillFee: Fee,
+    isBuy: boolean,
+  ): {
+    marginDelta: BigNumber,
+    positionDelta: BigNumber,
+  } {
+
+    const positionAmount = new BigNumber(fillAmount).dp(0, BigNumber.ROUND_DOWN);
+    const positionDelta = isBuy ? positionAmount.negated() : positionAmount;
```

Comment:
nit: might be slightly more readable to have `positionDelta` down by `marginDelta`, or to not have either constant and instead in-line it in the return statement since that's the only place either are used